core/src/test/java/org/apache/druid/data/input/impl/InputEntityIteratingReaderTest.java

```diff
@@ -95,4 +97,33 @@ public void test() throws IOException
       Assert.assertEquals(numFiles, i);
     }
   }
+
+  @Test
+  public void testIncorrectURI() throws IOException, URISyntaxException
```

Comment:
does this test issue an http request to `HTTP://test/path`? can we use a mock entity that throws an exception instead?